File: Backend/budget-service/src/main/java/com/finance/budget/model/Budget.java
package com.finance.budget.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.time.LocalDate;

import com.fasterxml.jackson.annotation.JsonFormat;

@Data
@Entity
@NoArgsConstructor
@AllArgsConstructor
@ToString
@Table(name = "budgets")
public class Budget {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Long userId;
    private String username;
    private String email;
    @JsonFormat(pattern = "yyyy-MM-dd")
    private LocalDate budgetStartDate;
    @JsonFormat(pattern = "yyyy-MM-dd")
    private LocalDate budgetEndDate;    
    private Double food;  
    private Double housing;  
    private Double transportation; 
    private Double entertainment;  
    private Double total;  
    private Boolean aiGenerated;
}

File: Frontend/finwiz/src/app/shared/models/budget.ts
export interface Budget {
    id: number;
    userId: number;
    username: string;
    email:string;
    budgetStartDate: Date;
    budgetEndDate: Date;
    food: number;
    housing: number;
    transportation: number;
    entertainment: number;
    aiGenerated: boolean;
    total:number;
  }
  